lib/dlopen.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+const binding = process.binding('dlopen');
+
+exports.dlopen = function dlopen(name) {
+  var lib = new Buffer(binding.sizeof_uv_lib_t);
+
+  if (binding.dlopen(name, lib) !== 0) {
+    // error
+    throw new Error(exports.dlerror(lib));
+  }
+
+  return lib;
+};
+
+exports.dlclose = function dlclose(lib) {
+  return binding.dlclose(lib);
+};
+
+exports.dlsym = function dlsym(lib, name) {
+  // TODO: use `sizeof.pointer` for buffer size when nodejs/io.js#1759 is merged
+  var sym = new Buffer(8);
```

Comment:
It just ends up containing the memory address of the symbol. You basically always do a `readPointer(0, byteLength)` call on this resulting buffer. You wouldn't do any damage altering the contents of this buffer, just change the memory of the buffer contents itself which is pretty useless.